rawkv/rawkv.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2021 TiKV Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// NOTE: The code in this file is based on code from the
+// TiDB project, licensed under the Apache License v 2.0
+//
+// https://github.com/pingcap/tidb/tree/cc5e161ac06827589c4966674597c137cc9e809c/store/tikv/rawkv.go
+//
+
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rawkv
+
+import (
+	"context"
+
+	"github.com/tikv/client-go/v2/config"
+	"github.com/tikv/client-go/v2/tikv"
+	pd "github.com/tikv/pd/client"
+)
+
+// Client is a client of TiKV server which is used as a key-value storage,
+// only GET/PUT/DELETE commands are supported.
+type Client struct {
+	client *tikv.RawKVClient
+}
+
+// NewClient creates a client with PD cluster addrs.
+func NewClient(ctx context.Context, pdAddrs []string, security config.Security, opts ...pd.ClientOption) (*Client, error) {
+	client, err := tikv.NewRawKVClient(pdAddrs, security, opts...)
+	if err != nil {
+		return nil, err
+	}
+	return &Client{client: client}, nil
+}
+
+// Close closes the client.
+func (c *Client) Close() error {
+	return c.client.Close()
+}
+
+// ClusterID returns the TiKV cluster ID.
+func (c *Client) ClusterID() uint64 {
+	return c.client.ClusterID()
+}
+
+// Get queries value with the key. When the key does not exist, it returns `nil, nil`.
+func (c *Client) Get(ctx context.Context, key []byte) ([]byte, error) {
+	return c.client.Get(key)
```

Comment:
The `ctx` is not used in the function. How can user cancel this execution using this `ctx`?

Reply:
I think it's remained for future work...

Reply:
Well then...

Reply:
Could we add some TODO to the function signatures in case we forget to refactor these codes? @disksing

Reply:
won't forget but I think it is better to add TODO :)